app/components/forms/wizard/sessions-speakers-step.js

```diff
@@ -175,19 +175,35 @@ export default Component.extend(EventWizardMixin, FormMixin, {
   },
 
   actions: {
-    addItem(type, position) {
+    addItem(type) {
       switch (type) {
         case 'sessionType':
           this.data.sessionTypes.addObject(this.store.createRecord('session-type'));
           break;
         case 'track':
           this.data.tracks.addObject(this.store.createRecord('track'));
           break;
-        case 'microlocation':
-          this.data.microlocations.addObject(this.store.createRecord('microlocation', { position }));
-          break;
       }
     },
+
+    addRoom(index) {
+      this.microlocations.forEach(room => {
+        const pos = room.get('position');
+        pos > index && room.set('position', pos + 1);
+      });
+      this.data.microlocations.addObject(this.store.createRecord('microlocation', { position: index + 1 }));
+    },
+
+    removeRoom(room, index) {
+      room.deleteRecord();
+      this.data.microlocations.forEach(item => {
+        const pos = item.get('position');
+        if (pos > index) {
+          item.set('position', pos - 1);
+        }
+      });
```

Comment:
Why need to shift positions?

Reply:
The same logic is acting on deleting action as well as inserting action.